pkg/quotaplugins/quota-forest/quota-manager/quota/quotamanager.go

```diff
@@ -372,9 +376,35 @@ func (m *Manager) AllocateForest(forestName string, consumerID string) (response
 	if consumerInfo == nil {
 		return nil, fmt.Errorf("consumer %s does not exist, create and add first", consumerID)
 	}
-	if forestController.IsConsumerAllocated(consumerID) {
-		return nil, fmt.Errorf("consumer %s already allocated on forest %s", consumerID, forestName)
+	//allocation can pass if consumers inside the forest is not released
```

Comment:
why do this whole logic when something like this might be more effective:
```golang
if forestController.IsConsumerAllocated(consumerID) {
		if !forestController.DeAllocate(consumerID) {
			return nil, fmt.Errorf("consumer %s already allocated on forest %s, failed to de-allocate", consumerID, forestName)
		}
	}
```

Reply:
How does this check if AW exists in the system?

Reply:
I am not sure if it matters: If the AWS is no longer in the system the quota should be released when the AWS is released.

Reply:
that does not happen, hence this PR